packages/core/src/storages/request_queue_v2.ts

```diff
@@ -164,6 +175,8 @@ export class RequestQueue extends RequestProvider {
             // Try to delete the request lock if possible
             try {
                 await this.client.deleteRequestLock(request.id!, { forefront: options?.forefront ?? false });
+
+                this.assumedForefrontCount++;
```

Comment:
This looks like an increment that should only happen in some cases. Missing ternary here?

Reply:
Good point; this uncovered another issue with the `MemoryStorage`'s implementation of RQv2 👍🏽 Added relevant tests and will solve as a part of this PR.